### PR TITLE
Added docs and unit test for zframe_meta()

### PR DESCRIPTION
**Pull request by petehayes102**
Added missing unit test and docs from #1389.

### CODE REVIEW

File: src/zframe.c [hunk 1]
```diff
@@ -571,9 +571,9 @@ zframe_test (bool verbose)
 
     //  @selftest
     //  Create two PAIR sockets and connect over inproc
-    zsock_t *output = zsock_new_pair ("@inproc://zframe.test");
+    zsock_t *output = zsock_new_pair ("@tcp://127.0.0.1:9001");
```
Comment (bluca) [inline src/zframe.c]: Why this change?

**Comment by bluca** [comment]
```
src/zframe.c: In function ‘zframe_meta’:  CC     src/src_libczmq_la-zhashx.lo
src/zframe.c:236:5: error: implicit declaration of function ‘zmq_msg_gets’ [-Werror=implicit-function-declaration]
src/zframe.c:236:5: error: return makes pointer from integer without a cast [-Werror]
```

File: src/zframe.c (hunk 1 above)
Comment (petehayes102) [inline src/zframe.c]: Argh...I stuffed that up, sorry. I was going to use zauth to return some meta data I could test, which doesn't work on inproc as far as I know. I didn't end up doing that, but now that I've reverted it to inproc, I'm getting a segfault... I'll need a few minutes to figure it out. I'm not very experienced with C.

**Comment by petehayes102** [comment]
I compiled this on OS X and didn't see those errors. I'll take a look.

File: src/zframe.c (hunk 1 above)
Comment (petehayes102) [inline src/zframe.c]: Ok, I got a segfault because I was running streq() on NULL. Duh!

Metadata doesn't get passed over inproc. Not sure why.

Comment (bluca) [inline src/zframe.c]: Ah ok, makes sense, thanks.

**Comment by petehayes102** [comment]
@bluca I just did a build on a fresh CentOS machine and wasn't able to reproduce the errors you got. Could you give me some more info about your platform?

I am compiling against ZMQ 4.1.4. What ZMQ version are you using?

I'm not sure why the compiler is complaining about an "implicit declaration" either. It seems all the right headers are there: zframe.c => czmq_classes.h => czmq.h => czmq_library.h => zmq.h => ZMQ_EXPORT const char *zmq_msg_gets

Any ideas?

**Comment by bluca** [comment]
So if you check on the CI build: https://travis-ci.org/zeromq/czmq/builds/122511912
The builds with older versions of ZMQ are failing. That's because the API you are using was introduced in 4.1. So, your new API should check the version like others are doing before using it, with \#ifdefs and so on. You can find a lot of examples in the code. Would you be able to send another PR with a fix?

I'll merge this in the meanwhile given it's not related to this changes.

**Comment by petehayes102** [comment]
Thanks. Will do.

**Comment by hintjens** [comment]
Yes, metadata isn't passed over inproc because it's generated by
authentication, and inproc connections don't do authentication.

**Comment by bluca** [comment]
2 more comments:

The api file is not only a documentation, is also the definition use by zproject, the code generator, to generate the API headers and stubs. After adding a new API there, the generation should be re-run and the result checked in. You can find more information here: https://github.com/zeromq/zproject

TL;DR: compile and install gsl, install zproject, run gsl project.xml in the czmq tree.

Then, for future contributions, please use the following format for the git commit:

```
Problem: brief one-line description of the problem you are solving

Solution: details of your solution
```

**Comment by petehayes102** [comment]
RE: Commit messages - I'm an idiot. I meant to do it for this PR and then went on autopilot. Sorry about that.

RE: API - Did you want me to PR the new zproject output along with the version gate stuff?

**Comment by bluca** [comment]
No worries!

If you could, that would be great. Feel free to do it separately, and ask for help for zproject if you need (learning curve can be a bit steep).
